finance/server/src/main/java/com/google/finapp/SpannerDaoPGImpl.java

```diff
@@ -0,0 +1,367 @@
+import com.google.cloud.spanner.Key;
+import com.google.cloud.spanner.KeySet;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.Statement;
+import com.google.cloud.spanner.TransactionContext;
+import com.google.cloud.spanner.Value;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.ByteString;
+import io.grpc.Status;
+import io.grpc.StatusException;
+import java.math.BigDecimal;
+
+final class SpannerDaoPGImpl implements SpannerDaoInterface {
+
+  private final DatabaseClient databaseClient;
+
+  SpannerDaoPGImpl(DatabaseClient databaseClient) {
+    this.databaseClient = databaseClient;
+  }
+
+  public void createCustomer(ByteArray customerId, String name, String address)
+      throws StatusException {
+    try {
+      this.databaseClient
+          .readWriteTransaction()
+          .run(
+              transaction -> {
+                Statement statement =
+                    Statement.newBuilder(
+                            "INSERT INTO Customer\n"
+                                + "(CustomerId, Name, Address)\n"
+                                + "VALUES\n"
+                                + "(@customerId, @name, @address)")
+                        .bind("customerId")
+                        .to(customerId)
+                        .bind("name")
+                        .to(name)
+                        .bind("address")
+                        .to(address)
+                        .build();
+                transaction.executeUpdate(statement);
+                return null;
+              });
+    } catch (SpannerException e) {
+      throw Status.fromThrowable(e).asException();
+    }
+  }
+
+  public void createAccount(ByteArray accountId, AccountStatus accountStatus, BigDecimal balance)
+      throws StatusException {
+    try {
+      this.databaseClient
+          .readWriteTransaction()
+          .run(
+              transaction -> {
+                Statement statement =
+                    Statement.newBuilder(
+                            "INSERT INTO Account\n"
+                                + "(AccountId, AccountStatus, Balance, CreationTimestamp)\n"
+                                + "VALUES\n"
+                                + "($1, $2, $3, SPANNER.PENDING_COMMIT_TIMESTAMP())")
+                        .bind("p1")
+                        .to(accountId)
+                        .bind("p2")
+                        .to(accountStatus.getNumber())
+                        .bind("p3")
+                        .to(
+                            Value.pgNumeric(
+                                balance.toString())) // Numeric requires special handling
+                        .build();
+                transaction.executeUpdate(statement);
+
+                return null;
+              });
+    } catch (SpannerException e) {
+      System.out.println(e.getMessage());
```

Comment:
Here and other methods: do we need these? I'd imagine the error would be logged upstream (in workload) or returned to the client via grpc. If we do, I'd recommend using 
import java.util.logging.Logger instead.

Reply:
No, this was put in during testing and not removed. Although, adding proper logging should be a future enhancement IMO

Reply:
This has been addressed in new commit